executor/executor_test.go

```diff
@@ -2645,6 +2645,51 @@ func (s *testSuite) TestCheckIndex(c *C) {
 	// table     data (handle, data): (1, 10), (2, 20), (4, 40)
 }
 
+func (s *testSuite) TestRenameIndex(c *C) {
+	s.ctx = mock.NewContext()
+	s.ctx.Store = s.store
+	_, err := session.BootstrapSession(s.store)
+	c.Assert(err, IsNil)
+	se, err := session.CreateSession4Test(s.store)
+	c.Assert(err, IsNil)
+	defer se.Close()
+
+	_, err = se.Execute(context.Background(), "create database test_rename_index")
+	c.Assert(err, IsNil)
+	_, err = se.Execute(context.Background(), "use test_rename_index")
+	c.Assert(err, IsNil)
+	_, err = se.Execute(context.Background(), "create table t (pk int primary key, c int default 1, c1 int default 1, unique key k1(c), key k2(c1))")
+	c.Assert(err, IsNil)
+
+	// Test rename success
+	_, err = se.Execute(context.Background(), "alter table t rename index k1 to k3")
+	c.Assert(err, IsNil)
+	_, err = se.Execute(context.Background(), "admin check index t k3")
+	c.Assert(err, IsNil)
+
+	// Test rename to the same name
+	_, err = se.Execute(context.Background(), "alter table t rename index k3 to k3")
+	c.Assert(err, IsNil)
+	_, err = se.Execute(context.Background(), "admin check index t k3")
+	c.Assert(err, IsNil)
+
+	// Test rename on non-exists keys
+	_, err = se.Execute(context.Background(), "alter table t rename index x to x")
+	c.Assert(err, NotNil)
+	c.Assert(strings.Contains(err.Error(), "Key 'x' doesn't exist in table 't'"), IsTrue)
```

Comment:
You can use `testErrorCode` in this file.

Reply:
I'll move these test to `ddl/ddl_db_test.go`